// @1inch/cross-chain-sdk/dist/cjs/api/relayer/relayer.request.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RelayerRequestSvm = exports.RelayerRequestEvm = void 0;
const chains_1 = require("../../chains");
class RelayerRequestEvm {
    constructor(params) {
        this.order = params.order;
        this.signature = params.signature;
        this.quoteId = params.quoteId;
        this.extension = params.extension;
        this.srcChainId = params.srcChainId;
        this.secretHashes = params.secretHashes;
    }
    build() {
        return {
            order: this.order,
            signature: this.signature,
            quoteId: this.quoteId,
            extension: this.extension,
            srcChainId: this.srcChainId,
            secretHashes: this.secretHashes
        };
    }
}
exports.RelayerRequestEvm = RelayerRequestEvm;
class RelayerRequestSvm {
    constructor(params) {
        this.order = params.order;
        this.quoteId = params.quoteId;
        this.secretHashes = params.secretHashes;
    }
    build() {
        return {
            order: this.order,
            quoteId: this.quoteId,
            srcChainId: chains_1.NetworkEnum.SOLANA,
            secretHashes: this.secretHashes
        };
    }
}
exports.RelayerRequestSvm = RelayerRequestSvm;
