// @1inch/cross-chain-sdk/dist/esm/api/relayer/relayer.request.js
import { NetworkEnum } from '../../chains';
export class RelayerRequestEvm {
    constructor(params) {
        this.order = params.order;
        this.signature = params.signature;
        this.quoteId = params.quoteId;
        this.extension = params.extension;
        this.srcChainId = params.srcChainId;
        this.secretHashes = params.secretHashes;
    }
    build() {
        return {
            order: this.order,
            signature: this.signature,
            quoteId: this.quoteId,
            extension: this.extension,
            srcChainId: this.srcChainId,
            secretHashes: this.secretHashes
        };
    }
}
export class RelayerRequestSvm {
    constructor(params) {
        this.order = params.order;
        this.quoteId = params.quoteId;
        this.secretHashes = params.secretHashes;
    }
    build() {
        return {
            order: this.order,
            quoteId: this.quoteId,
            srcChainId: NetworkEnum.SOLANA,
            secretHashes: this.secretHashes
        };
    }
}
